src/storage/page_buffer.c

```diff
@@ -8017,7 +8016,23 @@ pgbuf_read_page_from_file_or_page_server (THREAD_ENTRY * thread_p, const VPID *
     }
   if (request_from_ps)
     {
-      pgbuf_request_data_page_from_page_server (vpid);
+      // *INDENT-OFF*
+      const auto replicator_lsa_ftor = [] ()
+      {
+	const passive_tran_server *const pts_ptr = get_passive_tran_server_ptr ();
+	return pts_ptr->get_replicator_lsa ();
+      };
+      // *INDENT-ON*
+
+      /* on the passive transaction server, make sure to request that the progress on the page server - from
+       * where the data page is retrieved -  be at least the same as the local progress to avoid losing
+       * any log entries being applied on both ends
+       */
+      const LOG_LSA target_repl_lsa =
+	(is_passive_transaction_server ())? replicator_lsa_ftor () : pgbuf_Pool.get_highest_evicted_lsa ();
```

Comment:
- The lambda is overkill in this case. The code is easier to read without it.
- Complete the comment with active transaction server case.
```suggestion
      /* on the passive transaction server, make sure to request that the progress on the page server - from
       * where the data page is retrieved -  be at least the same as the local progress to avoid losing
       * any log entries being applied on both ends
       *
       * on the active transaction server, use highest evicted LSA to make sure no changes are missed by the PS.
       */
      const LOG_LSA target_repl_lsa =
	is_passive_transaction_server () ? get_passive_tran_server_ptr ()->get_replicator_lsa () : pgbuf_Pool.get_highest_evicted_lsa ();
```

Reply:
Done